### PR TITLE
refactor(tokens): break out AssetTabBar and AssetList into its own components

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

Break out AssetTabBar and AssetList from the Assets screen into its own components. This makes it easier to reason about and also would allow us to reuse components for the Wallets tab screen

### Test plan

Unit and manual tested

### Related issues

- Relates to ACT-1104

### Backwards compatibility

Yes

### Network scalability

N/A

### CODE REVIEW

File: src/tokens/AssetList.test.tsx
Comment (satish-ravi) [inline src/tokens/AssetList.test.tsx:1]: This is just a subset of tests from Assets.test.tsx

File: src/tokens/AssetList.tsx
Comment (satish-ravi) [inline src/tokens/AssetList.tsx:1]: Entirely copied from Assets.tsx

File: src/tokens/AssetTabBar.tsx
Comment (satish-ravi) [inline src/tokens/AssetTabBar.tsx:1]: Copied from Assets.tsx

File: src/tokens/PositionItem.tsx [hunk 1]
```diff
@@ -77,94 +73,19 @@ export const PositionItem = ({ position }: { position: Position }) => {
   )
 }
 
-export const TokenBalanceItem = ({
```
Comment (satish-ravi) [inline src/tokens/PositionItem.tsx:80]: No longer needed used after #5016 , knip didn't catch it because it was referenced in tests

**Review by silasbw — APPROVED** [review]

**Pull request merged**